if.go

```diff
@@ -46,29 +49,14 @@ func New(config Config) (ifce *Interface, err error) {
 	if zeroConfig == config {
 		config = defaultConfig()
 	}
-	return newDev(config)
-}
-
-// NewTAP creates a new TAP interface whose name is ifName. If ifName is empty, a
-// default name (tap0, tap1, ... ) will be assigned. ifName should not exceed
-// 16 bytes. TAP interfaces are not supported on darwin.
-// ifName cannot be specified on windows, you will need ifce.Name() to use some cmds.
-//
-// Note: this function is deprecated and will be removed from the library.
-// Please use New() instead.
-func NewTAP(ifName string) (ifce *Interface, err error) {
```

Comment:
I'm relatively new to Go, so I wasn't "around" before vendoring, but that makes sense.

The problem I foresee is that by using the `Name` param on a `Config` struct will cause platforms which don't have the name param to fail, which I suspected was the reason they were "deprecated" in favor of using `New`:
`config.Name undefined (type Config has no field or method Name)`

Of course we could work around that by adding `if_linux.go` with those functions. And if we're actually looking to deprecate those functions, perhaps print a warning message on use?